lib/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+### lib/CMakeLists.txt ###
+#
+#
+
+## Platform and system libraries
+include(platform_libs.cmake)
+
+## imgui:
+execute_process(
+	COMMAND git submodule update --init imgui
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_library(imgui imgui/imgui.cpp imgui/imgui_draw.cpp)
+target_include_directories(imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/imgui")
+target_link_libraries(imgui ${PLATFORM_LIBS})
+target_compile_definitions(imgui PUBLIC "-DImDrawIdx=unsigned int") # short is not enough for us (long causes core dumps)
+set(LIBS ${LIBS} imgui)
+
```

Comment:
I may have missed something but I don't think `imgui` is being added to `INCLUDE_DIRS`. Also I prefer exporting variables like `IMGUI_INCLUDE_DIRS` and adding them separaty with `target_include_directories` on the relevent target.